### PR TITLE
chore: Don't add conflicting labels to dependabot PRs

### PR DESCRIPTION
**Pull request by NotMyFault**
Title says all, this is a follow up to https://github.com/jenkinsci/jenkins/pull/6782

Dependabot rebases PRs automatically if they conflict or if we request a rebase, no need to add this label here.

### Proposed changelog entries

* Entry 1: Issue, Human-readable Text
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6798"></a>

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
tested somewhere to make sure the if is right?

**Comment by NotMyFault** [comment]
> tested somewhere to make sure the if is right?

According to the docs this we can use that for dependabot: https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#accessing-and-using-event-properties

**Review by timja — APPROVED** [review]

File: .github/workflows/label-conflicting-pr.yml [hunk 1]
```diff
@@ -3,12 +3,13 @@ on:
...
   pull_request_target:
     types: [synchronize]
-    
+
 permissions:
   pull-requests: write
 
 jobs:
   main:
+    if: github.event.pull_request.user.login != 'dependabot[bot]'
```
Comment (jglick) [inline .github/workflows/label-conflicting-pr.yml:12]: (For my edification: the `'` here is part of the value interpreted by GH, not a YAML metacharacter.)

**Review by basil — APPROVED** [review]
De minimis

**Comment by basil** [comment]
Fix failed: https://github.com/jenkinsci/jenkins/pull/6809

**Comment by NotMyFault** [comment]
Huh? Interesting, I'll need to take a look again. There may be something wrong with the docs 🤔

**Comment by davorpa** [comment]
> Huh? Interesting, I'll need to take a look again. There may be something wrong with the docs 🤔

Have you tried test value against `github.actor` instead of current condition?

**Pull request merged**